orca/orca.go

```diff
@@ -80,11 +80,15 @@ func unaryInt(ctx context.Context, req interface{}, _ *grpc.UnaryServerInfo, han
 	// We cannot call CallMetricRecorderFromContext() here to get the recorder
 	// from the context because that would lead to the allocation of a metric
 	// recorder if one wasn't allocated already.
-	recorderPointer, ok := ctxWithRecorder.Value(callMetricRecorderCtxKey{}).(**CallMetricRecorder)
-	if !ok || (*recorderPointer == nil) {
+	//
+	// It is safe to access the underlying metric recorder inside the wrapper at
+	// this point, as the user's RPC handler is done executing, and therefore
+	// there will be no more calls to CallMetricRecorderFromContext().
+	rw, ok := ctxWithRecorder.Value(callMetricRecorderCtxKey{}).(*recorderWrapper)
```

Comment:
It's a little awkward that we have to find this in the context considering we just stored it there.

Reply:
Hmm ... it is. But is there another option?

Reply:
I didn't want to prescribe one, and I also didn't want to suggest it since you just changed the function signature the other way, but....:

```go
rw := newRecorderWrapper()
ctxWithRecorder := contextWithRecorder(ctx, rw)
...
// use rw directly
```

Reply:
Done. Thanks.